hpe3parclient/client.py

```diff
@@ -508,46 +508,66 @@ def createVolume(self, name, cpgName, sizeMiB, optional=None):
 
         """
         info = {'name': name, 'cpg': cpgName, 'sizeMiB': sizeMiB}
-        # For primera array there is no compression and tdvv fields
-        # removing tdvv and replacing compression with reduce field
+        # For primera array there is no compression and tdvv keys
+        # removing tdvv, compression and
+        # replacing compression+tdvv with reduce key for DECO
         if not optional and self.primera_supported:
-            optional = {}
-            optional['tpvv'] = True
+            optional = {'tpvv': True}
         if optional:
             if self.primera_supported:
-                if 'tdvv' in optional:
-                    optional.pop('tdvv')
-
-                if optional.get('tpvv') is True \
+                for key in ['tpvv', 'compression', 'tdvv']:
+                    option = optional.get(key)
+                    if option and option not in [True, False]:
+                        # raising exception for junk compression input
+                        ex_desc = "39 - invalid input: wrong type for key " \
+                            "[%s], Valid values are[True, False]" % key
+                        raise exceptions.HTTPBadRequest(ex_desc)
+
+                combination = ['tdvv', 'compression']
+                len_diff = len(set(combination) - set(optional.keys()))
+                if len_diff == 1:
+                    raise exceptions.HTTPBadRequest(
+                        "Invalid input:For Deco volumes, "
+                        "both 'tdvv' and 'compression' must be specified")
```

Comment:
3PAR and Merlin's convention seems to be to use "invalid input:...." instead of "Invalid input:...".
Let's follow the lower case convention here as well.

Reply:
done